File: problem/Removing_Stars_String/removeStars.js
/**
 * @param {string} s
 * @return {string}
 */
var removeStars = function(s) {
    let c = "";
    for(let i = 0; i < s.length; i++){
        if(s[i] === '*'){
            c = c.slice(0, -1);
        }else{
            c += s[i];
        }
    }
    return c;
};

File: problem/Removing_Stars_String/removeStars.c
char *removeStars(char *s)
{
    int len = strlen(s);
    char *c = (char *)malloc((len + 1) * sizeof(char));
    int j = 0;
    for (int i = 0; i < len; i++)
    {
        if (s[i] == '*')
        {
            j--;
        }
        else
        {
            c[j++] = s[i];
        }
    }
    c[j] = '\0';
    return c;
}